cmd/buildctl/build.go

```diff
@@ -315,7 +316,21 @@ func buildAction(clicontext *cli.Context) error {
 }
 
 func writeMetadataFile(filename string, exporterResponse map[string]string) error {
-	b, err := json.Marshal(exporterResponse)
+	out := make(map[string]interface{})
+	for k, v := range exporterResponse {
+		dt, err := base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			out[k] = v
+			continue
+		}
+		var raw json.RawMessage
+		if err = json.Unmarshal(dt, &raw); err != nil {
```

Comment:
`RawMessage.UnmarshalJSON()` doesn't actually parse any JSON https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/encoding/json/stream.go;l=271 . This seems to still work but I'm not sure how 🤔 (or maybe just small inputs work) . But I think anyway it is safer to try parsing it to actual map interface. I think it is enough if we validate that it is a JSON object and don't handle liternals and arrays. That should minimize any possibility of randomly hitting this. Eg. "MTIK" is a valid base64 encoded JSON. (You can actually add it as a test case).

If the parsing is ok then use the data as `RawMessage` and discard the parsing result.

Reply:
>  it is a JSON object 

Lets also make sure it is a non-zero length JSON object.

Reply:
^